src/include/executor/index_scan_executor.h

```diff
@@ -88,6 +105,8 @@ class IndexScanExecutor : public AbstractScanExecutor {
   std::vector<expression::AbstractExpression *> runtime_keys_;
 
   bool key_ready_ = false;
+
+  planner::IndexScanPlan *plan_node_;
```

Comment:
I don't think we should use another plan_node_ here for IndexScanPlan, since we already have a node_ in the base class. If you already have the UpdatePredicate() method, can you call that method using node_ instead of have another non-const copy of the plan?